airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py

```diff
@@ -85,18 +89,41 @@ def __get__(self, instance, owner):
         return self._metadata
 
 
+class GoogleAnalyticsDataApiBackoffStrategy(BackoffStrategy):
+    def backoff_time(
+        self, response_or_exception: Optional[Union[requests.Response, requests.RequestException]], **kwargs: Any
+    ) -> Optional[float]:
+        if isinstance(response_or_exception, requests.Response):
+            # handle the error with prepared GoogleAnalyticsQuotaHandler backoff value
+            if response_or_exception.status_code == requests.codes.too_many_requests:
+                return GoogleAnalyticsQuotaHandler.backoff_time
+        return None
+
+
+class GoogleAnalyticsDatApiErrorHandler(HttpStatusErrorHandler):
+    @GoogleAnalyticsQuotaHandler.handle_quota()
+    def interpret_response(self, response_or_exception: Optional[Union[requests.Response, Exception]] = None) -> ErrorResolution:
+        if response_or_exception.status_code == requests.codes.too_many_requests:
+            return ErrorResolution(
+                response_action=GoogleAnalyticsQuotaHandler.response_action,
+                failure_type=FailureType.transient_error,
+                error_message=GoogleAnalyticsQuotaHandler.error_message,
+            )
+        return super().interpret_response(response_or_exception)
+
+
 class GoogleAnalyticsDataApiAbstractStream(HttpStream, ABC):
     url_base = "https://analyticsdata.googleapis.com/v1beta/"
     http_method = "POST"
-    raise_on_http_errors = True
+    _config: Mapping[str, Any] = {}
```

Comment:
Could you please leave the comment, why do we need this?

Reply:
At some stage of my development when running the UT I had an error message stating it couldn't resolve "_config" but I just removed this line and the error is not present so I can just assume it was something different that was causing that message, anyway will remove this and resolve the rest of your comments. Thanks!

Reply:
Fixed